code/datums/flock/flockprotection.dm

```diff
@@ -1,76 +1,83 @@
-			return
-		boutput(user, "<span class='alert'>The grip tool refuses to harm this, jamming briefly.</span>")
-		return TRUE
-
-	if (!istype(user, /mob/living/critter/flock/drone) && report_hand_attack)
-		src.attempt_report_attack(source, user)
-
-/datum/component/flock_protection/proc/handle_attackby(source, obj/item/W as obj, mob/user as mob)
-	if (istype(user, /mob/living/critter/flock/drone) && !flockdrones_can_hit)
-		if (istype(source, /obj/lattice/flock))
-			if (isweldingtool(W) && W:try_weld(user,0,-1,0,0))
-				boutput(user, "<span class='alert'>The grip tool refuses to allow damage to this, jamming briefly.</span>")
-				return TRUE
-		else if (istype(source, /obj/storage/closet/flock))
-			var/obj/storage/closet/flock/flockcloset = source
-			if (!flockcloset.open)
-				boutput(user, "<span class='alert'>The grip tool refuses to allow damage to this, jamming briefly.</span>")
-				return TRUE
-		else if (istype(source, /obj/machinery/door/feather))
-			var/mob/living/critter/flock/drone/flockdrone = user
-			if (flockdrone.find_type_in_hand(/obj/item))
-				boutput(user, "<span class='alert'>The grip tool refuses to allow damage to this, jamming briefly.</span>")
-				return TRUE
-		else
-			boutput(user, "<span class='alert'>The grip tool refuses to allow damage to this, jamming briefly.</span>")
-			return TRUE
-
-	if (!istype(user, /mob/living/critter/flock/drone) && report_obj_attack)
-		if (istype(source, /obj/lattice/flock))
-			if (isweldingtool(W) && W:try_weld(user,0,-1,0,0))
-				src.attempt_report_attack(source, user)
-		else if (istype(source, /obj/storage/closet/flock))
-			var/obj/storage/closet/flock/flockcloset = source
-			if (!flockcloset.open)
-				if (istype(W, /obj/item/cargotele) || istype(W, /obj/item/satchel))
-					return
-				else
-					src.attempt_report_attack(source, user)
-		else if (istype(source, /obj/machinery/door/feather))
-			if (user.equipped())
-				src.attempt_report_attack(source, user)
-		else
-			src.attempt_report_attack(source, user)
-
-/datum/component/flock_protection/proc/attempt_report_attack(source, mob/attacker)
-	var/list/nearby_flockdrones = list()
-	for (var/mob/living/critter/flock/drone/flockdrone in view(7, source))
-		if (!isdead(flockdrone) && flockdrone.is_npc && flockdrone.flock)
-			nearby_flockdrones.Add(flockdrone)
-
-	if (length(nearby_flockdrones))
-		var/mob/living/critter/flock/drone/flockdrone = pick(nearby_flockdrones)
-
-		if (!flockdrone.flock.isEnemy(attacker))
-			flock_speak(flockdrone, "Damage sighted on [source], [pick_string("flockmind.txt", "flockdrone_enemy")] [attacker]", flockdrone.flock)
-			flockdrone.flock.updateEnemy(attacker)
+/// Protect against someone shooting the parent.
+/datum/component/flock_protection/proc/handle_hitby(atom/source, obj/projectile/P)
+	var/mob/attacker = P.shooter
+	return istype(attacker) && src.report_thrown && SEND_SIGNAL(source, COMSIG_FLOCK_ATTACK, attacker, FALSE)
```

Comment:
```suggestion
	var/attacker = P.shooter
	return attacker && src.report_thrown && SEND_SIGNAL(source, COMSIG_FLOCK_ATTACK, attacker, FALSE)
```
Since vehicles can now be attackers